src/core-recipe.jl

```diff
@@ -86,22 +106,22 @@ function Makie.plot!(p::TopoPlot)
     end
     notify(p.resolution) # trigger above (we really need `update=true` for onany)
 
-    padded_data = lift(pad_data, p.data, npositions, p.pad_value)
-
-    if p.interpolation[] isa DelaunayMesh
-        # TODO, delaunay works very differently from the other interpolators, so we can't switch interactively between them
-        m = lift(delaunay_mesh, padded_position)
-        mesh!(p, m, color=padded_data, colorrange=p.colorrange, colormap=p.colormap, shading=false)
-    else
-        data = lift(p.interpolation, xg, yg, padded_position, padded_data) do interpolation, xg, yg, points, data
-            return interpolation(xg, yg, points, data)
-        end
-        heatmap!(p, xg, yg, data, colormap=p.colormap, colorrange=p.colorrange, interpolate=true)
-        contours = to_value(p.contours)
-        attributes = @plot_or_defaults contours Attributes(color=(:black, 0.5), linestyle=:dot, levels=6)
-        if !isnothing(attributes)
-            contour!(p, xg, yg, data; attributes...)
+    data = lift(p.interpolation, xg, yg, bounded, geometry) do interpolation, xg, yg, (points, data), geometry
+        z = interpolation(xg, yg, points, data)
+        if geometry isa Circle
+            c = geometry.center
+            r = geometry.r
+            out = [norm(Point2f(x, y) - c) > r for x in xg, y in yg]
+            z[out] .= NaN
         end
+        return z
+    end
+
+    heatmap!(p, xg, yg, data, colormap=p.colormap, colorrange=p.colorrange, interpolate=true)
```

Comment:
@SimonDanisch thoughts/feelings on using a heatmap instead of a mesh?

Reply:
We can use either:
1. `heatmap` uses the same gridded data we generate for the `contour` plot. Reusing the same data is cheap and also means that the contour lines and the heatmap's colors match perfectly (cause it's literally based on the same data).
2. `mesh` requires Delaunay triangulation to find the coordinates that best work with the a) original coordinates and b) the coordinates of the perimeter (i.e. `geometry`). We then need to interpolate the data for the coordinates of the triangles in that Delaunay mesh. There will probably be a slight mismatch between the resulting contour lines and color gradient, but I doubt it'll be noticeable?

The advantage of a `mesh` over a `heatmap` is that the edges of a mesh are "vectorized" while the colormap will be pixelated. So if you zoom in on the edge of a circle, you'll see the individual pixels of the heatmap (and the will depend on `resolution`), while in a mesh it'll just be a line (connecting the number of angles you used to decompose said circle, e.g. 64).

Reply:
I think a `heatmap` is fine for the moment, but we can always revisit this decision later.